packages/react-next/src/components/ResizeGroup/ResizeGroup.base.tsx

```diff
@@ -308,142 +308,194 @@ const hiddenDivStyles: React.CSSProperties = { position: 'fixed', visibility: 'h
-                {onRenderData(dataToMeasure)}
-              </MeasuredContext.Provider>
-            </div>
-          )}
-
-          <div
-            ref={this._initialHiddenDiv}
-            style={isInitialMeasure ? hiddenDivStyles : undefined}
-            data-automation-id="visibleContent"
-          >
-            {isInitialMeasure ? onRenderData(dataToMeasure) : renderedData && onRenderData(renderedData)}
-          </div>
-        </div>
-      </div>
-    );
+  function remeasure(): void {
+    if (rootRef.current) {
+      setMeasureContainer(true);
+    }
   }
 
-  public componentDidMount(): void {
-    this._afterComponentRendered(this.props.direction);
-    this._events.on(window, 'resize', this._async.debounce(this._onResize, RESIZE_DELAY, { leading: true }));
-  }
+  return [stateRef, updateResizeState, remeasure] as const;
+}
 
-  // tslint:disable-next-line function-name
-  public UNSAFE_componentWillReceiveProps(nextProps: IResizeGroupProps): void {
-    this.setState({
-      dataToMeasure: { ...nextProps.data },
-      resizeDirection: 'grow',
-      // Receiving new props means the parent might rerender and the root width/height might change
-      measureContainer: true,
-    });
-  }
+function useResizingBehavior(props: IResizeGroupProps, rootRef: React.RefObject<HTMLDivElement | null>) {
+  const nextResizeGroupStateProvider = useConst(getNextResizeGroupStateProvider);
 
-  public componentDidUpdate(prevProps: IResizeGroupProps) {
-    if (this.state.renderedData) {
-      this._hasRenderedContent = true;
-      if (this.props.dataDidRender) {
-        this.props.dataDidRender(this.state.renderedData);
-      }
-    }
-    this._afterComponentRendered(this.props.direction);
-  }
+  // A div that can be used for the initial measurement so that we can avoid mounting a second instance
+  // of the component being measured for the initial render.
+  const initialHiddenDiv = React.useRef<HTMLDivElement | null>(null);
+  // A hidden div that is used for mounting a new instance of the component for measurement in a hidden
+  // div without unmounting the currently visible content.
+  const updateHiddenDiv = React.useRef<HTMLDivElement | null>(null);
 
-  public componentWillUnmount(): void {
-    this._async.dispose();
-    this._events.dispose();
-  }
+  // Tracks if any content has been rendered to the user. This enables us to do some performance optimizations
+  // for the initial render.
+  const hasRenderedContent = React.useRef(false);
+
+  const async: Async = useAsync();
 
-  public remeasure(): void {
-    if (this._root.current) {
-      this.setState({ measureContainer: true });
+  const [stateRef, updateResizeState, remeasure] = useResizeState(props, nextResizeGroupStateProvider, rootRef);
+
+  React.useEffect(() => {
+    if (stateRef.current.renderedData) {
+      hasRenderedContent.current = true;
+      props.dataDidRender?.(stateRef.current.renderedData);
     }
-  }
+  });
```

Comment:
isn't this dependent on dataDidRender? I tjhink you need a deps list.

Reply:
For `useEffect`, the deps list controls how frequently the effect is run, not memoization as in `useCallback`. I’m not sure we want to prevent this from running every render.

Reply:
For `useEffect`, the deps list controls how frequently the effect is run, not memoization as in `useCallback`. I’m not sure we want to prevent this from running every render.

Reply:
For `useEffect`, the deps list controls how frequently the effect is run, not memoization as in `useCallback`. I’m not sure we want to prevent this from running every render.